src/handler.cr

```diff
@@ -5,33 +5,43 @@ require "./users_collection"
 
 # JWT authorization middleware
 class KemalJWTAuth::Handler < Kemal::Handler
-  # Used to encrypt all JWTs
-  private property secret_key : String
-  @logger : Logger
   property sign_in_endpoint = "/sign_in"
   property users : UsersCollection
+  property logger : Logger = Logger.new STDOUT
+  property algorithm : JWT::Algorithm = JWT::Algorithm::HS256
+  # Used to encrypt all JWTs
+  @secret_key : String
 
-  def initialize(@users : UsersCollection, @secret_key = Random::Secure.base64(32), @algorithm = "HS256", @logger = Logger.new STDOUT); end
+  # def initialize(@users, @secret_key : String, @algorithm : JWT::Algorithm, @logger : Logger)
+  # end
+
+  # def self.new(users_file : String, secret_key = Random::Secure.base64(32), algorithm = JWT::Algorithm::HS256::HS256, logger = Logger.new STDOUT)
+  #  File.open users_file do |file|
+  #    users = UsersCollection.from_json file
+  #  end
+  #  new users, secret_key, algorithm, logger
+  # end
 
   # Load in the users from the given location, where they are in JSON format.
-  # users_file may be any IO such as an open file or IO::Memory, or the string
-  # contents, or a string represeting the filepath at which the data is located.
-  def initialize(users_file : IO | String, @secret_key = Random::Secure.base64(32), @algorithm = "HS256", @logger = Logger.new STDOUT)
-    if users_file.is_a? String
-      filepath = users_file.as String
-      if File.exists? filepath
-        File.open filepath do |file|
-          @users = UsersCollection.from_json file
-        end
-        return
-      end
+  def initialize(@users : UsersCollection, @secret_key : String = Random::Secure.base64(32))
+  end
+
+  # :ditto:
+  def self.new(users_file : Path, secret_key : String = Random::Secure.base64(32))
+    File.open users_file.to_s do |file|
+      users = UsersCollection.from_json file
     end
-    @users = UsersCollection.from_json users_file
+    new users, secret_key
+  end
+
+  # :ditto:
+  def self.new(users_content : IO | String, secret_key : String = Random::Secure.base64(32))
+    new UsersCollection.from_json(users_io), secret_key
```

Comment:
I think you meant to use `users_content` in this call? Regardless, I'd prefer

```crystal
  def self.new(users content : IO | String, secret_key : String = Random::Secure.base64(32))
    new UsersCollection.from_json(content), secret_key
end
```
Same with the other `self.new` overload -- `users file` instead of `users_file`.